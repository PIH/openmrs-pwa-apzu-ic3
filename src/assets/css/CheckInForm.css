.panel {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 20px;
    background: rgba(255, 255, 255, 0.4);
    border-radius: 0px;
}

.midPanelItemContainer {
    margin-top: 20px;
    width: 220px;
}

.bottomPanelItemContainer {
    margin-top: 20px;
    width: 220px;
    align-content: stretch;
}

.panelItem {
    width: 220px;
}

.checkinButton {
    width: 100%;
    border-radius: 0px;
}

.form-control {
    background-color: #ffffff;
    border-radius: 0px;
    border: 0px;
    color: #777;
}

.referralSelector {
    background-color: #ffffff;
    height: 34px;
    width: 100%;
    padding: 6px 12px;
    font-size: 14px;
    line-height: 1.42857143;
    color: #555;
}
