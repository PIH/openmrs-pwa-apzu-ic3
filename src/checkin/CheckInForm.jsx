import React from 'react';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import CompletedScreenings from "../screening/CompletedScreenings";
import { FieldInput } from '@openmrs/react-components';
import { Alert, Button, ButtonToolbar, Grid, Row, Col, Form, FormGroup, ControlLabel, Label } from 'react-bootstrap';
import { goBack } from 'connected-react-router';


let CheckinForm = props => {

  const { handleSubmit, submitting, patient } = props;

  const historyBack = () => {
    props.dispatch(goBack());
  }

  return (
    <div>
      <Button bsSize='large' bsStyle='danger' onClick={historyBack.bind(this)}>
        Back
      </Button>
      <h3><Label>Check-in</Label></h3>
      <Form
        horizontal
        onSubmit={handleSubmit}
      >
        <Grid>

          { (typeof patient !== 'undefined') &&
            (typeof patient.alert !== 'undefined') &&
            <Row>
              <FormGroup controlId="formAlert">
                <Col
                  componentClass={ControlLabel}
                  sm={2}
                >
              Alert
                </Col>
                <Col
                  sm={4}
                >
                  <Alert bsStyle="danger">
                    { patient.alert }
                  </Alert>
                </Col>
              </FormGroup>
            </Row>
          }

          { (typeof patient !== 'undefined') &&
            (typeof patient.actions !== 'undefined') && (patient.actions !== patient.alert) &&
          <Row>
            <FormGroup controlId="formAction">
              <Col
                componentClass={ControlLabel}
                sm={2}
              >
              Action
              </Col>
              <Col sm={4}>
                <Alert bsStyle="warning">
                  { patient.actions }
                </Alert>
              </Col>
            </FormGroup>
          </Row>
          }

          <Row>
            <FormGroup controlId="formFirstName">
              <Col
                componentClass={ControlLabel}
                sm={2}
              >
                First Name
              </Col>
              <Col sm={4}>
                <Field
                  component={FieldInput}
                  name="name.givenName"
                  placeholder="First Name"
                  type='text'
                />
              </Col>
            </FormGroup>
          </Row>

          <Row>
            <FormGroup controlId="formLastName">
              <Col
                componentClass={ControlLabel}
                sm={2}
              >
                Last Name
              </Col>
              <Col sm={4}>
                <Field
                  component={FieldInput}
                  name="name.familyName"
                  placeholder="Last Name"
                  type='text'
                />
              </Col>
            </FormGroup>
          </Row>

          <Row>
            <FormGroup controlId="formGender">
              <Col
                componentClass={ControlLabel}
                sm={2}
              >
                Gender
              </Col>
              <Col sm={4}>
                <Field
                  component={FieldInput}
                  name="gender"
                  placeholder="Gender"
                  type='text'
                />
              </Col>
            </FormGroup>
          </Row>

          <Row>
            <FormGroup controlId="formAge">
              <Col
                componentClass={ControlLabel}
                sm={2}
              >
                Age
              </Col>
              <Col sm={4}>
                <Field
                  component={FieldInput}
                  name="age"
                  placeholder="Age"
                  type='number'
                />
              </Col>
            </FormGroup>
          </Row>


          {!(patient && patient.visit && patient.visit.encounters) &&
          <Row>
            <FormGroup controlId="formSubmit">
              <Col
                sm={4}
                smOffset={2}
              >
                <ButtonToolbar>
                  <Button
                    bsSize="large"
                    bsStyle="success"
                    disabled={submitting}
                    type="submit"
                  >
                    Check-in
                  </Button>

                </ButtonToolbar>
              </Col>
            </FormGroup>
          </Row>
          }

          {(patient && patient.visit && patient.visit.encounters) &&
            <div>
              <h3><Label>Completed Screenings</Label></h3>
              <CompletedScreenings/>
            </div>
          }
        </Grid>
      </Form>
    </div>
  );
};


CheckinForm = reduxForm({
  form: 'checkInForm', // a unique identifier for this form
})(CheckinForm);

CheckinForm = connect(
  state => ({
    patient: state.selectedPatient ? state.patients[state.selectedPatient] : null,
  })
)(CheckinForm);

export default CheckinForm;

