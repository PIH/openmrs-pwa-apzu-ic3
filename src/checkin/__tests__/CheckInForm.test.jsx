import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import MockRouter from 'react-mock-router';
import CheckInForm from '../CheckInForm';

let props, store;
let mountedComponent;

const mockStore = configureMockStore();

const checkInForm = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <Provider store={store}>
        <MockRouter>
          <CheckInForm {...props} />
        </MockRouter>
      </Provider>);
  }
  return mountedComponent;
};

describe('Component: CheckInForm', () => {
  beforeEach(() => {
    props = {
      patient: {
        uuid: "abc",
        name: {
          givenName: "Claire",
          familyName: "Jones"
        },
        age: 12,
        gender: "M"
      },
      backLink: "/checkin/checkInQueue"
    };
    store = mockStore({ openmrs: {} });
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    expect(toJson(checkInForm())).toMatchSnapshot();
  });

});
