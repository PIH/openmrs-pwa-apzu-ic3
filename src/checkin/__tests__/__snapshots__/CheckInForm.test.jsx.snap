// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CheckInForm renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MockRouter
    createHref={[Function]}
    location=""
    params={Object {}}
    path=""
    push={[Function]}
    replace={[Function]}
    url=""
  >
    <ReduxForm
      patient={
        Object {
          "age": 12,
          "gender": "M",
          "name": Object {
            "familyName": "Jones",
            "givenName": "Claire",
          },
          "uuid": "abc",
        }
      }
    >
      <Connect(Form(CheckinForm))
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="checkInForm"
        getFormState={[Function]}
        keepDirtyOnReinitialize={false}
        patient={
          Object {
            "age": 12,
            "gender": "M",
            "name": Object {
              "familyName": "Jones",
              "givenName": "Claire",
            },
            "uuid": "abc",
          }
        }
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      >
        <Form(CheckinForm)
          anyTouched={false}
          array={
            Object {
              "insert": [Function],
              "move": [Function],
              "pop": [Function],
              "push": [Function],
              "remove": [Function],
              "removeAll": [Function],
              "shift": [Function],
              "splice": [Function],
              "swap": [Function],
              "unshift": [Function],
            }
          }
          arrayInsert={[Function]}
          arrayMove={[Function]}
          arrayPop={[Function]}
          arrayPush={[Function]}
          arrayRemove={[Function]}
          arrayRemoveAll={[Function]}
          arrayShift={[Function]}
          arraySplice={[Function]}
          arraySwap={[Function]}
          arrayUnshift={[Function]}
          asyncValidating={false}
          autofill={[Function]}
          blur={[Function]}
          change={[Function]}
          clearAsyncError={[Function]}
          clearFields={[Function]}
          clearSubmit={[Function]}
          clearSubmitErrors={[Function]}
          destroy={[Function]}
          destroyOnUnmount={true}
          dirty={false}
          dispatch={[Function]}
          enableReinitialize={false}
          focus={[Function]}
          forceUnregisterOnUnmount={false}
          form="checkInForm"
          getFormState={[Function]}
          initialize={[Function]}
          initialized={false}
          invalid={false}
          keepDirtyOnReinitialize={false}
          patient={
            Object {
              "age": 12,
              "gender": "M",
              "name": Object {
                "familyName": "Jones",
                "givenName": "Claire",
              },
              "uuid": "abc",
            }
          }
          persistentSubmitErrors={false}
          pristine={true}
          pure={true}
          registerField={[Function]}
          reset={[Function]}
          resetSection={[Function]}
          setSubmitFailed={[Function]}
          setSubmitSucceeded={[Function]}
          shouldAsyncValidate={[Function]}
          shouldError={[Function]}
          shouldValidate={[Function]}
          shouldWarn={[Function]}
          startAsyncValidation={[Function]}
          startSubmit={[Function]}
          stopAsyncValidation={[Function]}
          stopSubmit={[Function]}
          submit={[Function]}
          submitFailed={false}
          submitSucceeded={false}
          submitting={false}
          syncErrors={Object {}}
          syncWarnings={Object {}}
          touch={[Function]}
          touchOnBlur={true}
          touchOnChange={false}
          unregisterField={[Function]}
          untouch={[Function]}
          updateSyncErrors={[Function]}
          updateSyncWarnings={[Function]}
          updateUnregisteredFields={false}
          valid={true}
          validExceptSubmit={true}
          values={Object {}}
        >
          <CheckinForm
            anyTouched={false}
            array={
              Object {
                "insert": [Function],
                "move": [Function],
                "pop": [Function],
                "push": [Function],
                "remove": [Function],
                "removeAll": [Function],
                "shift": [Function],
                "splice": [Function],
                "swap": [Function],
                "unshift": [Function],
              }
            }
            asyncValidate={[Function]}
            asyncValidating={false}
            autofill={[Function]}
            blur={[Function]}
            change={[Function]}
            clearAsyncError={[Function]}
            clearFields={[Function]}
            clearSubmit={[Function]}
            clearSubmitErrors={[Function]}
            destroy={[Function]}
            dirty={false}
            dispatch={[Function]}
            form="checkInForm"
            handleSubmit={[Function]}
            initialize={[Function]}
            initialized={false}
            invalid={false}
            patient={
              Object {
                "age": 12,
                "gender": "M",
                "name": Object {
                  "familyName": "Jones",
                  "givenName": "Claire",
                },
                "uuid": "abc",
              }
            }
            pristine={true}
            pure={true}
            reset={[Function]}
            resetSection={[Function]}
            submit={[Function]}
            submitFailed={false}
            submitSucceeded={false}
            submitting={false}
            touch={[Function]}
            untouch={[Function]}
            valid={true}
          >
            <div>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsSize="large"
                bsStyle="danger"
                disabled={false}
                onClick={[Function]}
              >
                <button
                  className="btn btn-lg btn-danger"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Back
                </button>
              </Button>
              <h3>
                <Label
                  bsClass="label"
                  bsStyle="default"
                >
                  <span
                    className="label label-default"
                  >
                    Check-in
                  </span>
                </Label>
              </h3>
              <Form
                bsClass="form"
                componentClass="form"
                horizontal={true}
                inline={false}
                onSubmit={[Function]}
              >
                <form
                  className="form-horizontal"
                  onSubmit={[Function]}
                >
                  <Grid
                    bsClass="container"
                    componentClass="div"
                    fluid={false}
                  >
                    <div
                      className="container"
                    >
                      <Row
                        bsClass="row"
                        componentClass="div"
                      >
                        <div
                          className="row"
                        >
                          <FormGroup
                            bsClass="form-group"
                            controlId="formSubmit"
                          >
                            <div
                              className="form-group"
                            >
                              <Col
                                bsClass="col"
                                componentClass="div"
                                sm={4}
                                smOffset={2}
                              >
                                <div
                                  className="col-sm-4 col-sm-offset-2"
                                >
                                  <ButtonToolbar
                                    bsClass="btn-toolbar"
                                  >
                                    <div
                                      className="btn-toolbar"
                                      role="toolbar"
                                    >
                                      <Button
                                        active={false}
                                        block={false}
                                        bsClass="btn"
                                        bsSize="large"
                                        bsStyle="success"
                                        disabled={false}
                                        type="submit"
                                      >
                                        <button
                                          className="btn btn-lg btn-success"
                                          disabled={false}
                                          type="submit"
                                        >
                                          Check-in
                                        </button>
                                      </Button>
                                    </div>
                                  </ButtonToolbar>
                                </div>
                              </Col>
                            </div>
                          </FormGroup>
                        </div>
                      </Row>
                    </div>
                  </Grid>
                </form>
              </Form>
            </div>
          </CheckinForm>
        </Form(CheckinForm)>
      </Connect(Form(CheckinForm))>
    </ReduxForm>
  </MockRouter>
</Provider>
`;
